### DISCUSSION TITLE
issue: select value will not update properly on firefox/chrome

### DISCUSSION THREAD

**Discussion by birdhackor** [react-hook-form/react-hook-form#7297] [Q&A]
### Version Number

7.22.1

### Codesandbox/Expo snack

https://codesandbox.io/s/busy-bash-wgrwe 

### Steps to reproduce

1. Open [codesandbox1](https://codesandbox.io/s/busy-bash-wgrwe) with Firefox 95.0 (Newest for now) or Chrome
2. Switch select box 1 **will not** change select box 2 value
3. Open same page with safari
4. Switch select box 1 **will** change select box 2 value
5. Modify code (option key)
6. Open [codesandbox2](https://codesandbox.io/s/awesome-http-rnsp4) with Firefox 95.0 (Newest for now) or Chrome
7. Behavior will be all the same

### Expected behaviour

Switch select box 1 **should** change select box 2 value when useing key with `value` only.

### What browsers are you seeing the problem on?

- Firefox
- Chrome

_Edge not tested_

### Code of Conduct

- [ ] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
```
const [options2] = useState(["1", "2"]);
const [options3] = useState(["2", "3"]);

{(getValues("select1") === "a" ? options2 : options3).map(
  (value, i) => (
    <option key={value}>{value}</option>
  )
)}
```

I tried with chrome and working fine, and the above usage is a bit strange as well.  if you just want to update other input, just simply add `onChange` with `setValue`.

**Reply by birdhackor** [reply]
That's strange... I've tested on newest Chrome on windows 10 and it's not working correctly.

On the other hand, use `setValue` still has same issue.

I modify code with it, and result as below:

https://codesandbox.io/s/admiring-engelbart-jw6ry

When you use firefox (and I test chrome on Win 10 too) to open it, select2 will be `2`.

But if you use safari (15.1) to open it, select2 will be `1`, and this is behavior I expect. 

Second version: https://codesandbox.io/s/distracted-roman-x884n

This version add `(typeof getValues("select1") === "undefined"` part.

When you open it, select2 will be `1` as expected.

But when you change select1 to `b` (select2 will auto switch to `2`), and then change select1 back to `a`, select2 remain to be `2`.

I except it to be `1`.  

Perfect version is here: https://codesandbox.io/s/lingering-thunder-bdt47

If change key from `key={value}` to ``key={{`${value}-${i}`}``, it will work perfectly.

(`(typeof getValues("select1") === "undefined"` part is used to avoid select2 blink to 1 from 2)

**Reply by birdhackor** [reply]
@bluebill1049 Do you have any suggestion?

**Comment by Moshyfawn** [comment]
I'd suggest resetting the dependent select on the actual form value change instead of the `onChange` callback of the main select, as the options of the dependent select wouldn't be the correct ones during the `onChange` callback execution, as the main select value hasn't been set yet.
That was a mouth full xD You can just take a look at [this CSB](https://codesandbox.io/s/dazzling-goodall-jicjs?file=/src/App.tsx)
